articles/mobile-automation-testing-with-appium/index.md

```diff
@@ -0,0 +1,99 @@
+[Image Source: Software Testing Help](https://www.softwaretestinghelp.com/what-is-appium/)
+
+The Appium architecture consists of three main components: the Appium client, Appium server, and the end device.
+
+#### Appium client
+This consists of the automation code (scripted). This code is written in a specific programming language (Java, Python, Ruby, or Perl). The script contains the configuration details of the application and mobile device. The scripted code and the configuration details are used for running the application’s test cases.
+
+#### Appium server
+This is an HTTP server that receives command requests (HTTP requests) from the client component (in JSON format). This server uses the Node.js language. The commands are then executed on mobile devices. This component should be installed on the computer before scripting the automation code. The Appium server acts as an intermediary between the Appium Client and the end device. It is connected to platforms such as iOS, Android, and Windows.
+
+#### End device
+This refers to an emulator or a mobile device that is connected to the Appium server. The test cases are executed on this device.
+
+### How Appium works?
+Appium works using the three main components of its architecture (client, server, and end device). When this tool is installed on your computer, it sets up an HTTP server that generates a REST API.
+
+This server collects command requests from the Appium client in JSON format. JSON Wire Protocol plays the role of communicating these requests. These command requests are then executed on the end devices (either Android, Windows, or iOS).
+
+The Appium server uses a test automation framework to execute requests on the user interface of end devices. For example, it uses *XCUI test* to execute commands on iOS. It uses *UI Automator* to execute requests on Android platforms (for newer API versions). After the requests have been executed, the Appium server provides HTTP responses to the client.
+
+Appium works differently on iOS and Android. Let's look at how it works on these two platforms.
+
+#### How Appium works on Android?
+![Appium on Android](/engineering-education/articles/mobile-automation-testing-with-appium/appium-on-android.png)
+[Image Source: Edureka](https://www.edureka.co/blog/content/ver.1556540029/uploads/2019/05/Appium-on-Android-Appium-Architecture-edureka-4.png)
+
+The following is a brief sequence of how Appium works on the Android platform:
+1.The Appium client connects with the HTTP server and uses the JSON Wire Protocol for communication.
+2.The Appium server receives requests from the client component.
+3.The server connects with the mobile automation framework for Android (UI Automator or Selendroid).
+4.Android devices consist of **bootstrap.jar** that plays the role of a [TCP server](https://en.wikipedia.org/wiki/Transmission_Control_Protocol). The **bootstrap.jar** receives the test commands from the Appium server.  

```

Comment:
A few suggestions:
- Explain what is the role of TCP server here.
- Also, explain what does `bootstrap.jar` contain.